lib/rubocop/cop/layout/extra_spacing_with_bindata_ignored.rb

```diff
@@ -0,0 +1,30 @@
+module RuboCop
+  module Cop
+    module Layout
+      class ExtraSpacingWithBinDataIgnored < ExtraSpacing
+
+        def_node_matcher :bindata?, <<~PATTERN
+          (class _ (const (const _ :BinData) _) _)
+        PATTERN
```

Comment:
Note: For now I'm ignoring the entire BinData class in its entirety, we could make it smarter in the future but it's a good first step

Rubocop matcher created from the documentation:
https://docs.rubocop.org/rubocop-ast/node_pattern.html

And seeing the sexpressions with:
```
ruby-parse -e 'class Foo < BinData::Record; end;'
(class
  (const nil :Foo)
  (const
    (const nil :BinData) :Record) nil)
```